subsystem/elevator.py

```diff
@@ -0,0 +1,74 @@
+import wpilib
+from robotpy_toolkit_7407 import Subsystem
+from robotpy_toolkit_7407.motors.revmotor import SparkMax
+from robotpy_toolkit_7407.utils.units import meters
+from robotpy_toolkit_7407.sensors.limit_switches.limit_switch import LimitSwitch
+
+import constants
+import config
+#importing packages
+
+
+class Elevator(Subsystem): #elevator class
+    extendMotor: SparkMax = SparkMax(config.elevator_motor_extend_id)#motor that extends the arm
+    rotationMotorRight: SparkMax = SparkMax(config.elevator_motor_turn_right_id, inverted = True) #motor that rotates the arm
+    rotationMotorLeft: SparkMax = SparkMax(config.elevator_motor_turn_left_id, inverted = False) #motor that rotates the arm
+    initialized: bool = False
+    extend_sensor = LimitSwitch(9) #senses when extension is at zero
+    turn_sensor = LimitSwitch(8) #senses when turn is at zero
+
+    def init(self): #initializing motors
+        self.extendMotor.init()
+        self.rotationMotorRight.init()
+        self.rotationMotorLeft.init()
+        
+    
+    def set_length(self, h: meters): #set arm extension
+        self.extendMotor.set_target_position(h * constants.elevator_gear_ratio)
+
+    def set_rotation(self, h: meters): #set arm rotation
+        self.rotationMotorRight.set_target_position(h * constants.elevator_gear_ratio)
+        self.rotationMotorLeft.set_target_position(h* constants.elevator_gear_ratio)
+
+    def get_length(self): #returns arm extension
+        return self.extendMotors.get_sensor_position() / constants.elevator_gear_ratio
+    
+    def get_rotation(self, h: meters): #returns arm rotation
+        return self.rotationMotorLeft.get_sensor_position() / constants.elevator_gear_ratio
+    
+
+    def zero_elevator(self): #brings elevator to zero position (no extension, no rotation)
+        while(self.extend_sensor.get_value() == False):
```

Comment:
While loops will stop the rest of the code from running while this is rezeroing. I will show you how to use commands tomorrow, which are a way to avoid this issue.

Reply:
I was planning on explaining our elevator rezero command from last year (which pretty much does what you say here) on Monday.